src/index.ts

```diff
@@ -4,22 +4,21 @@ dotenv.config();
 import { setupApp } from './app';
 import { close } from './services/client';
 import { initiateLogger } from './services/logger';
+import { Server } from 'http';
 
 const PORT = process.env.PORT || 3000;
 const logger = initiateLogger();
+let server: Server;
 
 const startServer = async () => {
-  const app = await setupApp({});
-
-  const server = app.listen(PORT, () => {
-    logger.info(`Server listening on port: ${PORT}`);
-  });
-
-  process.on('SIGINT', async () => {
-    logger.info('SIGINT signal received');
-    await close();
-    server.close();
-  });
+  try {
+    const app = await setupApp({});
+    server = app.listen(PORT, () => {
+      logger.info(`Server listening on port: ${PORT}`);
+    });
+  } catch (e) {
+    throw e;
+  }
```

Comment:
since we are not returning setupApp, i believe any error this throws will be uncaught.

small example here:

```
const setupApp = async () => {
  return new Promise((_res, rej) => {
    setTimeout(() => {
      rej(new Error('test error'))
    }, 1500)
  })
};


const run = async () => {
  await setupApp();
  console.log('run finish');
}

try {
  run()
} catch (e) {
  console.error('caught at 18 ', e);
}
```

results in
```
Desktop node test-promise.js             
(node:56334) UnhandledPromiseRejectionWarning: Error: test error
    at Timeout._onTimeout (/Users/seung.park/Desktop/test-promise.js:4:11)
    at listOnTimeout (internal/timers.js:557:17)
    at processTimers (internal/timers.js:500:7)
(Use `node --trace-warnings ...` to show where the warning was created)
(node:56334) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). To terminate the node process on unhandled promise rejection, use the CLI flag `--unhandled-rejections=strict` (see https://nodejs.org/api/cli.html#cli_unhandled_rejections_mode). (rejection id: 1)
(node:56334) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
```

Reply:
perhaps we should run with `--unhandled-rejections=strict` in the cmd to make sure the node process is terminated as noted

Reply:
Oh, forgot it was a promise. 🤦 In that case, wouldn't the error be caught if we used `.catch()` instead?

```js
run().catch((e) => {
  console.error('caught at 18 ', e);
});
```

Something like ^ would probably accomplish what we want without needing a redundant try-catch?

Reply:
sounds good! pushed this suggestion

Reply:
With the new change for catching, I think it's safe to remove this try-catch block now